### ISSUE TITLE
Facing issue with smbj file read with reloading file conitnuously

### ISSUE THREAD

**Issue by harisingh-highq** [hierynomus/smbj#348]
@hierynomus & @pepijnve 
As i have a remote log file reloads continuously with some new content adding into that file.
So with smbj program, i am trying to read that reload file but its not read the new content adding into that file. 
Here is code snippet attached:-
`try(Connection connection = client.connect("fileserver1ind1.hqdev.india"))
		{
			AuthenticationContext ac = new AuthenticationContext("jcifsuser", "H3lloworld".toCharArray(), "hqdev.india");
			session = connection.authenticate(ac);

			// Connect to Share
			try (DiskShare share = (DiskShare) session.connectShare("data"))
			{
				// File Read/download
				System.out.println("======File Read start=========");
				String remotefilePathForDownload = "Harisingh\\collab_4_1_4\\TestSMBUpload.txt";
				File smbFileDownload = share.openFile(remotefilePathForDownload, EnumSet.of(AccessMask.GENERIC_READ), null, SMB2ShareAccess.ALL,
						SMB2CreateDisposition.FILE_OPEN, null);
				java.io.InputStream is = smbFileDownload.getInputStream();
				BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(is));
				String output = bufferedReader.readLine();
				while (true)
				{
//					System.out.println(output);
					output = bufferedReader.readLine();
				}
				smbFileDownload.close();
				System.out.println("====== File Read End =========");			
			}
			catch (IOException e)
			{
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
		}`

**Comment by harisingh-highq** [comment]
@hierynomus 
Are you getting above question?

**Comment by neoxpert** [comment]
I guess the problem might be, that the underlying input stream is flagged as closed as soon as the first EOF was read ( com.hierynomus.smbj.share.FileInputStream.java line 133 ). Every consecutive read call will be returned without rechecking for file changes ( com.hierynomus.smbj.share.FileInputStream.java line 113) .

**Comment by harisingh-highq** [comment]
@neoxpert 
Yeah. That might be the cause. So you mean to say that right now, smbj api does not support consecutive calls reading/writing?

**Comment by neoxpert** [comment]
At least it seems like it is not supported by the default InputStream out of the box. But smbj offers all required functions to get this done.

**Comment by harisingh-highq** [comment]
> But smbj offers all required functions to get this done.

Can you please attach list of functions with some lines of code example to do that?
